### PR TITLE
bug fix & add router

### PR DESCRIPTION
**Pull request by realmorrisliu**

### CODE REVIEW

File: routes/messageRouter.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+var messageRouter = require("express").Router();
+
+var messageController = require('../controllers/messageController');
+
+messageRouter.get('/getting', messageController.getMessage);
+messageRouter.post('/sending', messageController.sendMessage);
```
Comment (williamyu-xingyunzh) [inline routes/messageRouter.js:6]: Seems you have not exported via "module.exports". So how can you get it via "require" in router.js?

File: routes/router.js [hunk 2]
```diff
@@ -10,6 +11,5 @@ module.exports = function(app, contextRoot) {
     rootRouter.post('/addDemo', demoController.addDemo);
 
     //Please make any business router under the rootRouter, so that it will be easy for contextRoot config.
-    rootRouter.get('/message', messageController.message);
-    rootRouter.post('/sendMessage', messageController.sendMessage);
+    rootRouter.use('/message');
```
Comment (williamyu-xingyunzh) [inline routes/router.js]: You need to associate messageRouter to this path.
say, rootRouter.use('/message', messageRouter);

File: routes/router.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 var rootRouter = require("express").Router();
+var messageRouter = require("./messageRouter");
 
 var demoController = require('../controllers/demoController');
 var messageController = require('../controllers/messageController');
```
Comment (williamyu-xingyunzh) [inline routes/router.js]: messageController is no longer needed here.  Let demoController be kept here. I will remove it eventually.

**Review by williamyu-xingyunzh — CHANGES_REQUESTED** [review]
You'd better unit test your change before submitting the request.
Btw, you can submit part of our work, but please ensure that is your intent, say adding "TODO","To be continued", or "TBD" comments there for unfinished part.

**Comment by realmorrisliu** [comment]
fixed

**Comment by williamyu-xingyunzh** [comment]
Hi Morris，
I'm not able to merge your pull request. because "This branch has conflicts that must be resolved".
You need update your branch with latest master's code, and resolve conflicts well.

**Comment by morrieati** [comment]
conflicts resolved

**Comment by realmorrisliu** [comment]
conflicts resolved

**Review by williamyu-xingyunzh — APPROVED** [review]
Since you've add a message page in ui, just keep it for test. Please touch /public/\* as litttle as possible. You can use Postman or other RESTful tools to test your backend service.

**Pull request merged**